src/index.js

```diff
@@ -190,12 +32,71 @@ export const stagePlugins = [
   "transform-object-rest-spread",
 ];
 
+/**
+ * Determine if a transformation is required
+ * @param  {Object}  supportedEnvironments  An Object containing environment keys and the lowest
+ *                                          supported version as a value
+ * @param  {Object}  plugin                 An Object containing environment keys and the lowest
+ *                                          version the feature was implmented in as a value
+ * @return {Boolean}  Whether or not the transformation is required
+ */
+export const isPluginRequired = (supportedEnvironments, plugin) => {
+  const targetEnvironments = Object.keys(supportedEnvironments);
+
+  if (targetEnvironments.length === 0) { return true; }
+
+  const isRequiredForEnvironments = targetEnvironments
+    .filter(environemt => {
+      // Feature is not implemented in that environment
+      if (!plugin[environemt]) { return true; }
+
+      const lowestImplementedVersion = plugin[environemt];
+      const lowestTargetedVersion = targetEnvironments[environemt];
+      if (lowestTargetedVersion <= lowestImplementedVersion) { return true; }
+
+      return false;
+    });
+
+  return isRequiredForEnvironments.length > 0 ? true : false;
+};
+
+const getTargets = targetOpts => {
+  return targetOpts || {};
+}
+
+// TODO: Allow specifying plugins as either shortened or full name
+// babel-plugin-transform-es2015-classes
+// transform-es2015-classes
+const getLooseMode = looseOpts => {
+  if (!looseOpts) { return []; }
+  return looseOpts;
+}
+
+// TODO: Allow specifying modules as: Boolean, String or Array of module types
+const getModules = modulesOpts => {
+  if (!modulesOpts) { return []; }
+  return modulesOpts;
+}
+
 export default function(opts) {
-  let stage = 4;
-  if (opts !== undefined){
-    if (opts.stage !== undefined) stage = opts.stage;
-  }
+  const looseMode = getLooseMode(opts.loose);
+  const modulesMode = getModules(opts.modules);
+  const targets = getTargets(opts.targets);
 
-  if (typeof stage !== "boolean") throw new Error("Preset es2015 'loose' option must be a boolean.");
+  const transformations = Object.keys(pluginList)
+    .filter(pluginName => isPluginRequired(targets, pluginList[pluginName]))
+    .map(pluginName => {
+      return includes(looseMode, pluginName)
+        ? [require(`babel-plugin-${pluginName}`), { loose }]
+        : require(`babel-plugin-${pluginName}`);
```

Comment:
I used the es2015 preset as an example here. If I don't need to do the full require or if it's okay to have an Array with just one element like `[require(`babel-plugin-${pluginName}`)]` it would clean up this statement a little bit.